File: mobile/ios/Handlers/ActionHandler.swift
import Foundation

final class ActionHandler {
    
    private let context: PaymentSDKPlugin
    
    init(context: PaymentSDKPlugin) {
        self.context = context
    }
    
    func execute(action: String, args: String) {
        let action = ActionEnum.fromString(action: action)
        let callback = EventEmitter(context: context, action: action)
        
        do {
            let paymentHandler = PaymentHandler(callback: callback)
            
            switch action {
            case .performTransaction:
                try paymentHandler.performTransaction(args: args)
            case .performEMVTransaction:
                try paymentHandler.performEMVTransaction(args: args)
            case .setDeviceType:
                try paymentHandler.setDeviceType(args: args)
            case .scanForDevices:
                try paymentHandler.scanForDevices(args: args)
            case .stopScanForDevices:
                try paymentHandler.stopScanForDevices()
            case .connectDeviceByName:
                try paymentHandler.connectDeviceByName(args: args)
            case .initialDeviceSetup:
                try paymentHandler.initialDeviceSetup(args: args)
            case .disconnectDevice:
                try paymentHandler.disconnectDevice()
            case .setTimeouts:
                try paymentHandler.setTimeouts(args: args)
            case .unknown:
                break
            }
        } catch let error {
            callback.sendEvent(
                type: .error,
                message: error.localizedDescription
            )
        }
    }
}

File: mobile/android/app/src/main/java/com/fortis/businessportal/sdk/ActionHandler.java
package com.sdkplayground.newarchitecture.sdk;

import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONArray;

public class ActionHandler {

    private final ReactApplicationContext reactContext;

    public ActionHandler(final ReactApplicationContext reactContext) {
        this.reactContext = reactContext;
    }

    public void execute(
            String action,
            JSONArray args) {

        ActionEnum actionEnum = ActionEnum.fromString(action);
        EventEmitter callbackUtils = new EventEmitter(reactContext, actionEnum);

        try {
            PaymentHandler paymentHandler = new PaymentHandler(reactContext, callbackUtils);

            switch (actionEnum) {
                case ACTION_PERFORM_TRANSACTION:
                    paymentHandler.performTransaction(args);
                    break;
                case ACTION_PERFORM_EMV_TRANSACTION:
                    paymentHandler.performEMVTransaction(args);
                    break;
                case ACTION_SET_DEVICE_TYPE:
                    paymentHandler.setDeviceType(args);
                    break;
                case ACTION_SCAN_FOR_DEVICES:
                    paymentHandler.scanForDevices(args);
                    break;
                case ACTION_STOP_SCAN_FOR_DEVICES:
                    paymentHandler.stopScanForDevices();
                    break;
                case ACTION_CONNECT_DEVICE_BY_NAME:
                    paymentHandler.connectDeviceByName(args);
                    break;
                case ACTION_INITIAL_DEVICE_SETUP:
                    paymentHandler.initialDeviceSetup(args);
                    break;
                case ACTION_DISCONNECT_DEVICE:
                    paymentHandler.disconnectDevice();
                    break;
                case ACTION_SET_TIMEOUTS:
                    paymentHandler.setTimeouts(args);
                    break;
                default:
                    throw new IllegalArgumentException("Action " + action + " is not currently being handled.");
            }
        } catch (Exception ex) {
            callbackUtils.error(ex.getMessage());
        }
    }
}